client/me/purchases/payment-methods/existing-credit-card.tsx

```diff
@@ -0,0 +1,406 @@
+	}
+`;
+
+const CardHolderName = styled.span`
+	display: block;
+`;
+
+function ExistingCardLabel( {
+	last4,
+	cardExpiry,
+	cardholderName,
+	brand,
+	storedDetailsId,
+	paymentPartnerProcessorId,
+}: {
+	last4: string;
+	cardExpiry: string;
+	cardholderName: string;
+	brand: string;
+	storedDetailsId: string;
+	paymentPartnerProcessorId: string;
+} ): JSX.Element {
+	const { __, _x } = useI18n();
+	const translate = useTranslate();
+
+	const [ isDialogVisible, setIsDialogVisible ] = useState( false );
+	const closeDialog = useCallback( () => setIsDialogVisible( false ), [] );
+	const queryClient = useQueryClient();
+
+	const { data } = useQuery<
+		{ tax_postal_code: string; tax_country_code: string; is_tax_info_set: boolean },
+		Error
+	>( [ 'tax-info-is-set', storedDetailsId ], () => fetchTaxInfo( storedDetailsId ), {} );
+
+	const mutation = useMutation(
+		( mutationInputValues: { tax_postal_code: string; tax_country_code: string } ) =>
+			setTaxInfo(
+				storedDetailsId,
+				mutationInputValues.tax_postal_code,
+				mutationInputValues.tax_country_code
+			),
+		{
+			onMutate: ( onMutateInputValues: { tax_postal_code: string; tax_country_code: string } ) => {
+				// Optimistically update the toggle
+				queryClient.setQueryData( [ 'tax-info-is-set', storedDetailsId ], {
+					tax_postal_code: onMutateInputValues.tax_postal_code,
+					tax_country_code: onMutateInputValues.tax_country_code,
+					is_tax_info_set: true,
+				} );
+				closeDialog();
+			},
+			onSuccess: ( onSuccessInputValues: {
+				tax_postal_code: string;
+				tax_country_code: string;
+			} ) => {
+				queryClient.setQueryData( [ 'tax-info-is-set', storedDetailsId ], {
+					tax_postal_code: onSuccessInputValues.tax_postal_code,
+					tax_country_code: onSuccessInputValues.tax_country_code,
+					is_tax_info_set: true,
+				} );
+			},
+		}
+	);
+
+	const [ inputValues, setInputValues ] = useState( {
+		tax_postal_code: data?.tax_postal_code ?? '',
+		tax_country_code: data?.tax_country_code ?? '',
+	} );
+
+	const postalCodeValue = inputValues.tax_postal_code;
+	const countryCodeValue = inputValues.tax_country_code;
+
+	const updateTaxInfo = useCallback( () => {
+		mutation.mutate( inputValues );
+	}, [ mutation, inputValues ] );
+
+	const onChangeCountryCode = ( e: { target: { value: string } } ) => {
+		setInputValues( { ...inputValues, tax_country_code: e.target.value } );
+	};
+
+	const onChangePostalCode = ( e: { target: { value: string } } ) => {
+		setInputValues( { ...inputValues, tax_postal_code: e.target.value } );
+	};
+
+	const renderEditForm = (): JSX.Element => {
+		return (
+			<form>
+				<div className="contact-fields payment-methods__tax-fields">
+					<RenderEditFormFields
+						postalCodeValue={ postalCodeValue }
+						countryCodeValue={ countryCodeValue }
+						onChangePostalCode={ onChangePostalCode }
+						onChangeCountryCode={ onChangeCountryCode }
+					/>
+				</div>
+			</form>
+		);
+	};
+
+	const formRender = renderEditForm();
```

Comment:
I'm not sure it's worth the extra indirection to make `renderEditForm` a function if we're not going to use it as a top-level component. I suggest doing one of two things here; either:

- Change this line to `const taxForm = ( <form><div className="contact-fields payment-methods__tax-fields"> ... </form )`
- Move `renderEditForm` outside of this component and rename it something like `function TaxForm({postalCodeValue, countryCodeValue, onChangePostalCode, onChangeCountryCode}) { ... }`.

I think the first option is probably better because `RenderEditFormFields` is already a component that looks very much like `renderEditForm`.